gprofiler/utils/perf.py

```diff
@@ -35,6 +38,72 @@
 )
 
 
+class SUPPORTED_PERF_EVENTS(Enum):  # pylint: disable=C0103
+    """
+    order here is crucial, the first one we try and succeed - will be used.
+    keep it mind that we should always use `PERF_DEFAULT` as a first try.
+    meaning - keeping with `perf`s' default is always preferred.
+    """
+
+    PERF_DEFAULT = "default"
+    PERF_SW_CPU_CLOCK = "cpu-clock"
+    PERF_SW_TASK_CLOCK = "task-clock"
+
+    def perf_extra_args(self) -> List[str]:
+        if self == SUPPORTED_PERF_EVENTS.PERF_DEFAULT:
+            return []
+        return ["-e", self.value]
+
+
+def get_perf_event_args(tmp_dir: Path, stop_event: Event) -> List[str]:
+    """
+    Get the list of arguments to use the appropriate event in `perf record`.
+
+    We've observed that on some machines the default event `perf record` chooses doesn't actually collect samples.
+    And we generally would not want to change the default event chosen by `perf record`, so before
+    any change we apply to collected sample event, we want to make sure that the default event
+    actually collects samples, and make changes only if it doesn't.
+
+    :param tmp_dir: working directory of this function
+    :return: `perf record` extra arguments to use (e.g. `["-e", "cpu-clock"]`)
+    """
+
+    for event in SUPPORTED_PERF_EVENTS:
+        perf_script_output = None
+
+        try:
+            current_extra_args = event.perf_extra_args() + [
+                "sleep",
+                "0.5",
+            ]  # `sleep 0.5` is enough to be certain some samples should've been collected.
+            perf_process = PerfProcess(
```

Comment:
Note on using `PerfProcess` here - it's slightly wasteful and slows down gProfiler boot, because `PerfProcess` works with continuous perf (`--switch-output`). We don't require that for the `sleep 0.5` short run we're trying to do, it only makes things longer by 1.5s per my measuring.
Please take a look if you can perhaps add code in `PerfProcess` that'll allow to control a perf started without this option (control it with a c'tor parameter?), then we don't need the `start` and `stop` and it can be just a single call to run it which returns with the `perf script` result.
If it's a mess to add support - let's consider skipping it (it is not a must improvement IMO).

Reply:
what makes you believe that this makes a difference?
will `--switch-output=signal` still have such overhead impact? if it won't, I think this approach is fine 2d72a63286a97125b3b6365cacefee4fd4607b77

Reply:
what makes you believe that this makes a difference?
will --switch-output=signal still have such overhead impact? if it won't, I think this approach is fine 2d72a63286a97125b3b6365cacefee4fd4607b77